dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskSolver.scala

```diff
@@ -1,13 +1,55 @@
 package io.joern.dataflowengineoss.queryengine
 
 import io.joern.dataflowengineoss.queryengine.QueryEngineStatistics.{PATH_CACHE_HITS, PATH_CACHE_MISSES}
+import io.joern.dataflowengineoss.queryengine.TaskSolver.{doneTaskCounter, totalTaskCounter}
 import io.joern.dataflowengineoss.semanticsloader.Semantics
 import io.shiftleft.codepropertygraph.generated.nodes._
 import io.shiftleft.semanticcpg.language.{toCfgNodeMethods, toExpressionMethods}
 
-import java.util.{Calendar, UUID}
-import java.util.concurrent.Callable
-import scala.collection.mutable
+import java.util.concurrent.{Callable}
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicInteger, AtomicLong}
+
+object TaskSolver {
+  val totalTaskCounter      = new AtomicInteger(0)
+  val doneTaskCounter       = new AtomicInteger(0)
+  val futuresStartedCounter = new AtomicInteger(0)
+  val futuresEndedCounter   = new AtomicInteger(0)
+  val lastDoneTasks         = new AtomicInteger(0)
+  val lastPrintTime         = new AtomicLong()
+  val myTurn                = new AtomicBoolean()
+
+  def printStats(): Unit = {
+
+    val now: Long = System.currentTimeMillis / 1000
+
+    if (now - lastPrintTime.get() < 1) {
+      return
+    }
+    lastPrintTime.set(now)
+
+    val prevVal = myTurn.getAndSet(true)
+    if (prevVal == true) {
+      return
+    }
+
+    val totalTasks     = totalTaskCounter.get()
+    val doneTasks      = doneTaskCounter.get()
+    val futuresStarted = futuresStartedCounter.get()
+    val futuresEnded   = futuresEndedCounter.get()
+    val backlog        = futuresStarted - futuresEnded
+
+    println(
```

Comment:
We need to convert these prints to trace logs.

Reply:
Converted to debug logs

Reply:
They print the progress and give visibility on what is happening. That is why I thought they should be debug logs. Please let me know if you still think they should be trace